Build Error: StandardPerformanceCounter IDisposable

Build system is throwing this error intermittently. Need to confirm this change is made in the code.

> ##[error]WEB\Src\PerformanceCollector\PerformanceCollector\Implementation\StandardPerformanceCollector\StandardPerformanceCounter.cs(12,36): 
> Error CA2213: 'StandardPerformanceCounter' contains field 'performanceCounter' that is of IDisposable type 'PerformanceCounter', but it is never disposed. 
> Change the Dispose method on 'StandardPerformanceCounter' to call Close or Dispose on this field.

--- a/.props/Product.props
+++ b/.props/Product.props
@@ -16,14 +16,14 @@
     <PackageReference Include="StyleCop.Analyzers" Version="1.1.118">
       <PrivateAssets>All</PrivateAssets>
     </PackageReference>
-    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="3.0.0">
+    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="3.3.0">
       <PrivateAssets>All</PrivateAssets>
     </PackageReference>
     <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.0">	
       <PrivateAssets>all</PrivateAssets>	
       <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>	
     </PackageReference>
-    <PackageReference Include="Microsoft.CodeAnalysis.PublicApiAnalyzers" Version="3.0.0">
+    <PackageReference Include="Microsoft.CodeAnalysis.PublicApiAnalyzers" Version="3.3.0">
       <PrivateAssets>All</PrivateAssets>
     </PackageReference>
   </ItemGroup>

--- a/.rulesets/ApplicationInsightsSDKRules.ruleset
+++ b/.rulesets/ApplicationInsightsSDKRules.ruleset
@@ -31,9 +31,12 @@
     <Rule Id="CA1410" Action="Warning" />
     <!-- Suppressing Message "Identifiers should not match keywords" because class TplActivities is part of our public api and can't be changed -->
     <Rule Id="CA1716" Action="None" />
+    <!-- Errors shown after FxCop upgrade to 3.3.0. Turning off CA1805 to ignore these errors-->
     <Rule Id="CA1805" Action="None" />
     <Rule Id="CA1821" Action="Warning" />
+    <!-- Errors shown after FxCop upgrade to 3.3.0. Turning off CA1829 to ignore these errors-->
     <Rule Id="CA1829" Action="None" />
+    <!-- Errors shown after FxCop upgrade to 3.3.0. Turning off CA1834 to ignore these errors-->
     <Rule Id="CA1834" Action="None" />
     <Rule Id="CA1900" Action="Warning" />
     <Rule Id="CA2001" Action="Warning" />

--- a/.rulesets/ApplicationInsightsSDKRules.ruleset
+++ b/.rulesets/ApplicationInsightsSDKRules.ruleset
@@ -31,13 +31,9 @@
     <Rule Id="CA1410" Action="Warning" />
     <!-- Suppressing Message "Identifiers should not match keywords" because class TplActivities is part of our public api and can't be changed -->
     <Rule Id="CA1716" Action="None" />
-    <!-- Errors shown after FxCop upgrade to 3.3.0. Turning off CA1805 to ignore these errors-->
+    <!-- Suppressing Message "Do not initialize unnecessarily" because this won't change performance -->
     <Rule Id="CA1805" Action="None" />
     <Rule Id="CA1821" Action="Warning" />
-    <!-- Errors shown after FxCop upgrade to 3.3.0. Turning off CA1829 to ignore these errors-->
-    <Rule Id="CA1829" Action="None" />
-    <!-- Errors shown after FxCop upgrade to 3.3.0. Turning off CA1834 to ignore these errors-->
-    <Rule Id="CA1834" Action="None" />
     <Rule Id="CA1900" Action="Warning" />
     <Rule Id="CA2001" Action="Warning" />
     <Rule Id="CA2002" Action="Warning" />

--- a/.rulesets/ApplicationInsightsSDKRules.ruleset
+++ b/.rulesets/ApplicationInsightsSDKRules.ruleset
@@ -31,7 +31,10 @@
     <Rule Id="CA1410" Action="Warning" />
     <!-- Suppressing Message "Identifiers should not match keywords" because class TplActivities is part of our public api and can't be changed -->
     <Rule Id="CA1716" Action="None" />
+    <Rule Id="CA1805" Action="None" />
     <Rule Id="CA1821" Action="Warning" />
+    <Rule Id="CA1829" Action="None" />
+    <Rule Id="CA1834" Action="None" />
     <Rule Id="CA1900" Action="Warning" />
     <Rule Id="CA2001" Action="Warning" />
     <Rule Id="CA2002" Action="Warning" />

--- a/BASE/src/Microsoft.ApplicationInsights/Metrics/Implementation/ConcurrentDatastructures/MultidimensionalCube2.cs
+++ b/BASE/src/Microsoft.ApplicationInsights/Metrics/Implementation/ConcurrentDatastructures/MultidimensionalCube2.cs
@@ -258,7 +258,11 @@
 
                 if (i > 0)
                 {
+#pragma warning disable CA1834 // Consider using 'StringBuilder.Append(char)' when applicable
+                    // TODO: check if we can change from string to char, this will have some work
+                    // since it's used as separator in split and contains
                     builder.Append(PointMonikerSeparator);
+#pragma warning restore CA1834 // Consider using 'StringBuilder.Append(char)' when applicable
                 }
 
                 builder.Append(coordinates[i]);

--- a/BASE/src/Microsoft.ApplicationInsights/Metrics/MetricIdentifier.cs
+++ b/BASE/src/Microsoft.ApplicationInsights/Metrics/MetricIdentifier.cs
@@ -640,14 +640,14 @@
                 idStr.Append(MetricIdentifier.NoNamespaceIdentifierStringComponent);
             }
             
-            idStr.Append("+");
+            idStr.Append('+');
             idStr.Append(this.MetricId);
 
-            idStr.Append("[");
+            idStr.Append('[');
             idStr.Append(this.DimensionsCount);
-            idStr.Append("]");
+            idStr.Append(']');
 
-            idStr.Append("(");
+            idStr.Append('(');
 
             for (int d = 1; d <= this.DimensionsCount; d++)
             {
@@ -661,7 +661,7 @@
                 idStr.Append('"');
             }
 
-            idStr.Append(")");
+            idStr.Append(')');
 
             return idStr.ToString();
         }

--- a/BASE/src/ServerTelemetryChannel/Implementation/TransmissionStorage.cs
+++ b/BASE/src/ServerTelemetryChannel/Implementation/TransmissionStorage.cs
@@ -269,11 +269,11 @@
 
         private void LoadFilesOrderedByDateFromFolder()
         {
-            if (this.files.Count == 0)
+            if (this.files.IsEmpty)
             {
                 lock (this.loadFilesLock)
                 {
-                    if (this.files.Count == 0)
+                    if (this.files.IsEmpty)
                     {
                         // Sleep a tiny bit before (re)loading the list so that any other thread still processing
                         // a file has time to finish and delete it so that it does not get re-added to the new list.

--- a/WEB/Src/DependencyCollector/DependencyCollector/Implementation/AzureSdk/AzureSdkDiagnosticsEventHandler.cs
+++ b/WEB/Src/DependencyCollector/DependencyCollector/Implementation/AzureSdk/AzureSdkDiagnosticsEventHandler.cs
@@ -369,7 +369,7 @@
                     linksJson.Remove(linksJson.Length - 1, 1);
                 }
 
-                linksJson.Append("]");
+                linksJson.Append(']');
                 telemetry.Properties["_MS.links"] = linksJson.ToString();
             }
         }

--- a/WEB/Src/PerformanceCollector/PerformanceCollector/PerformanceCollectorModule.cs
+++ b/WEB/Src/PerformanceCollector/PerformanceCollector/PerformanceCollectorModule.cs
@@ -272,7 +272,9 @@
                 stopwatch.Stop();
 
                 PerformanceCollectorEventSource.Log.CounterCollectionSuccessEvent(
+#pragma warning disable CA1829 // Use Length/Count property instead of Count() when available
                     results.LongCount(),
+#pragma warning restore CA1829 // Use Length/Count property instead of Count() when available
                     stopwatch.ElapsedMilliseconds);
 
                 foreach (var result in results)

--- a/WEB/Src/WindowsServer/WindowsServer/Implementation/MetricManager.cs
+++ b/WEB/Src/WindowsServer/WindowsServer/Implementation/MetricManager.cs
@@ -374,7 +374,7 @@ namespace Microsoft.ApplicationInsights.WindowsServer
                 aggregationIntervalDuation = aggregationPeriod;
             }
 
-            if (aggregatorSnapshot.Count > 0)
+            if (!aggregatorSnapshot.IsEmpty)
             {
                 foreach (KeyValuePair<Metric, SimpleMetricStatisticsAggregator> aggregatorWithStats in aggregatorSnapshot)
                 {

